### PR TITLE
Polymer upgrade

### PR DESCRIPTION
**Pull request by MrSerth**
Updating our second element with four embedded web component elements to Polymer 2.0.

// cc: @franzliedke

### CODE REVIEW

File: mammooc-course-evaluation-form.html [hunk 1]
```diff
@@ -58,41 +64,43 @@
...
+                <flexible-rating id="stars" name="stars" required></flexible-rating>
+                <paper-button class="submit-button" raised on-click="_submitForm">
+                    {{localize("submit")}}
+                </paper-button>
+            </form>
+        </iron-form>
 
         <!-- Send form if user is not logged in -->
-        <form method="post" target="_blank" action="https://mammooc.org/evaluations/login_and_save" id="nativeForm">
+        <form method="post" target="_blank" action="https://mammooc-dev.herokuapp.com/evaluations/login_and_save" id="nativeForm">
```
Comment (franzliedke) [inline mammooc-course-evaluation-form.html:103]: Is this change on purpose?

File: mammooc-course-evaluation-form.html [hunk 2]
```diff
@@ -106,132 +114,150 @@
...
-        },
-
-        _postSent: function(event) {
-            document.getElementById('evaluationsXHR').generateRequest();
-            this.$.successToast.open();
-        },
-
-        _submitForm: function() {
+            _postSent(event) {
+                this.parentNode.parentNode.parentNode.querySelector('#evaluationsXHR').generateRequest();
```
Comment (franzliedke) [inline mammooc-course-evaluation-form.html]: Hmm, this looks like it is very closely coupled to the current HTML structure. When somebody introduces a new HTML element in the tree or changes the structure slightly, this will just break...

File: mammooc-course-evaluation-form.html (hunk 1 above)
Comment (MrSerth) [reply mammooc-course-evaluation-form.html:103]: Yes, it is. Due to a mistake, the dev branch currently holds this "wrong" URL, which is only applicable to our master. I will double check the URLs prior to publishing a new release.

File: mammooc-course-evaluation-form.html (hunk 2 above)
Comment (MrSerth) [reply mammooc-course-evaluation-form.html]: You're absolutely right and I do not prefer the current solution. Unfortunately, I was not able yet to find another. Here's why - maybe you have an idea?:

`this`:
`<mammooc-course-evaluation-form>`

`this.parentNode`:
`<div class="card-actions">`

`this.parentNode.parentNode`:
`<paper-card elevation="1" aria-label="">`

`this.parentNode.parentNode.parentNode`:
`#shadow-root` (of the `mammooc-rating-widget`)

`this.parentNode.parentNode.parentNode.querySelector('#evaluationsXHR')`:
`<iron-ajax id="evaluationsXHR" auto="" hidden="">`

An alternative would be to use 
`document.querySelector('mammooc-rating-widget').shadowRoot.querySelector('#evaluationsXHR')` which leads to problems if multiple `mammooc-rating-widget`s are defined on a single page.

Unfortunately, I was not able to find a way using an "upwards" querySelector. `closest()`, for example, could be used to get the `paper-card`, but it doesn't the `mammooc-rating-widget` nor the corresponding shadow root.

**Review by janrenz — APPROVED** [review]

File: mammooc-course-evaluation-form.html (hunk 2 above)
Comment (franzliedke) [reply mammooc-course-evaluation-form.html]: @MrSerth Can't you give the `<mammooc-rating-widget>` element a method that returns the `evaluationsXHR` request - or better, encapsulates sending the request? Then, you only need to (dynamically) look upwards in the DOM for the next `mammooc-rating-widget` element?
